javaagent/build.gradle.kts

```diff
@@ -240,11 +242,30 @@ tasks {
     delete(rootProject.file("licenses"))
   }
 
+  val removeLicenseDate by registering {
+    // removing the license report date makes it idempotent
+    doLast {
+      val filePath = rootDir.toPath().resolve("licenses").resolve("licenses.md")
+      if (Files.exists(filePath)) {
+        val datePattern = Pattern.compile("^_[0-9]{4}-[0-9]{2}-[0-9]{2} [0-9]{2}:[0-9]{2}:[0-9]{2} .*_$")
+        val lines = Files.readAllLines(filePath).map {
+          if (datePattern.matcher(it).matches()) {
+            ""
+          } else {
+            it
+          }
+        }.toCollection(ArrayList<String>())
```

Comment:
The line with date is always 4th line in the file, I'd just check that line.
It is unfortunate that
```
  class LicenseReportRenderer : InventoryMarkdownReportRenderer() {
    override fun printHeader() {
      output.appendText("## Dependency License Report\n")
    }
  }
  renderers = arrayOf(LicenseReportRenderer())
```
we can't use. Apparently extending `InventoryMarkdownReportRenderer`, that is written in groovy, from kotlin won't work correctly.